spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaProducerFactory.java

```diff
@@ -92,7 +93,7 @@
 
 	private String transactionIdPrefix;
 
-	private volatile boolean running;
+	private AtomicBoolean running = new AtomicBoolean(true);
```

Comment:
This must be `final` and it can't be `true` by default. What if I'm not going to start this component?
Also this is a behavior change.

Please, reconsider this logic for proper expectations.

Reply:
I set it with `true` by default because running state checking added in `createProducer()`, make it auto started not throwing exception at first time, keep behavior consistency.

Reply:
I agree to remove `Lifecycle`